docker-bake.hcl

```diff
@@ -0,0 +1,292 @@
+variable "TAG" {
```

Comment:
I'd like to see this more heavily commented to help us better understand this; it should also be documented in the README.md or Development.md

Reply:
I added top level comments for the various variables, groups, and targets. Rather than copy / paste the vendor docs, I linked to the 2 main topics from the bake file. 

Side note; We're also adding this documentation in the work on #15